src/physics.cpp

```diff
@@ -420,63 +419,30 @@ void sample_positron_reaction(Particle* p)
   p->alive = false;
 }
 
-void sample_nuclide(const Particle* p, int mt, int* i_nuclide, int* i_nuc_mat)
+int sample_nuclide(const Particle* p)
 {
   // Sample cumulative distribution function
-  double cutoff;
-  switch (mt) {
-  case SCORE_TOTAL:
-    cutoff = prn() * simulation::material_xs.total;
-    break;
-  case SCORE_SCATTER:
-    cutoff = prn() * (simulation::material_xs.total -
-      simulation::material_xs.absorption);
-    break;
-  case SCORE_FISSION:
-    cutoff = prn() * simulation::material_xs.fission;
-    break;
-  }
+  double cutoff = prn() * simulation::material_xs.total;
```

Comment:
Oh good, why did we have the other options in the past anyways?

Reply:
The other options were eventually intended to be used for tallies (see the 6 year old issue #148), but that obviously hasn't gotten traction. Plus, this routine is called so often that it ought to be as simple as possible. If we do want to sample reactions based on cross sections other than total, it might be better to have a separate function anyway (or something templated).